Recruit-Training/Viscomm-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -7,31 +8,75 @@ namespace
 
 BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
 : initialStateOfChargePercent_(initialStateOfChargePercent)
+, currentData_(new BatteryData())
+, pastData_(new BatteryData())
+, totalCurrent_(0.0)
+, currentCount_(0)
+, totalAmpHours_(BATTERY_AMP_HOUR_CAPACITY * (initialStateOfChargePercent / 100))
 {
 }
 
 BatteryStateOfChargeService::~BatteryStateOfChargeService()
 {
+    delete pastData_;
+    delete currentData_;
+}
+
+void BatteryStateOfChargeService::updateAmpHours(){
+
+    if(pastData_->time.QTime::isNull()){
+        return;
+    }
+
+    double currentAvg = totalCurrent_ / currentCount_;
+    double timeDifference = pastData_->time.QTime::msecsTo(currentData_->time) / 3600000.0;
```

Comment:
All instances of `3600000.0` should be replaced with a constant so we can know that the number means.